kore/src/Kore/Logger/DebugSolver.hs

```diff
@@ -67,3 +92,44 @@ logDebugSolverRecvWith
     -> m ()
 logDebugSolverRecvWith logger smtText =
     logWith logger $ DebugSolverRecv smtText
+
+solverTranscriptLogger
+    :: Applicative m
+    => LogAction m Text
+    -> LogAction m SomeEntry
+solverTranscriptLogger textLogger =
+    LogAction
+    $ \entry -> case matchDebugSolverSend entry of
+        Just sendEntry -> unLogAction textLogger (messageToText sendEntry)
+        Nothing -> unLogAction mempty entry
+  where
+    messageToText :: DebugSolverSend -> Text
+    messageToText =
+        Pretty.renderStrict
+        . Pretty.layoutPretty Pretty.defaultLayoutOptions
+        . pretty
+
+
+matchDebugSolverSend :: SomeEntry -> Maybe DebugSolverSend
+matchDebugSolverSend = fromEntry
+
+{- | Options (from the command-line) specifying where to create a solver
+transcript.
+
+See also: 'parseDebugSolverOptions'
+
+-}
+newtype DebugSolverOptions =
+    DebugSolverOptions
+        { logType :: KoreLogType
```

Comment:
I'm confused about the `KoreLogType`. I thought that this would be something like `Maybe FilePath`: either the user requested the transcript in a particular file, or they did not. What am I missing?

Reply:
It allows me to provide a uniform interface for `startLoggers` in a natural way. Also, it seemed more flexible, but we don't actually need that. 

Since I can always do a `Maybe FilePaht` to `KoreLogType` conversion before calling `startLoggers`, I'll change this to `Maybe FilePath`.

Reply:
Fixed.